lstchain/scripts/lstchain_dl1ab.py

```diff
@@ -164,6 +164,7 @@ def main():
         nodes_keys.remove(dl1_images_lstcam_key)
 
     auto_merge_h5files([args.input_file], args.output_file, nodes_keys=nodes_keys)
```

Comment:
This is actually performed by the `smart_merge_h5files` rather than the `auto_merge_h5files` - and as a bonus, you get a quick check on the metadata to see if we are merging consistent files.

I think however that the `smart_merge_h5files` is causing issues when merging a lot of files if 
`metadata0.SOURCE_FILENAMES` gets too long. Here:
https://github.com/cta-observatory/cta-lstchain/blob/5032edb033a3d01bc6db45b930ad62d2df3a302a/lstchain/io/io.py#L329

I can't find again the limit (imposed by hdf5 I think) but a quick test gives me a limit at 65500 characters for a single attribute.
Maybe you can implement that limit and use the smart merge?

Reply:
I overlooked the fact that the `smart_merge` has been designed to merge MC files, while `dl1ab` is not limited to MC, so that won't work actually.

Reply:
Maybe you can just add the `check_metadata` before merging to make sure the file created is consistent?
But the `SOURCE_FILENAMES` issue remains, you are keeping only one filename here.

Reply:
@vuillaut 
I think in this case we are not merging two existing files where we can check if their metadata are equal, we are creating a new hdf5 file from content stored in another one and we want to recover also the metadata. Actually `smart_merge_h5files` receives as input a list of more than one already existing files. 

Maybe we could refactor a bit all these functions, i.e. add a call to metadata copying and/or checking  in `auto_merge_h5files`? and remove `smart_merge_h5files` having a unique *smart* function `auto_merge_h5files` to copy and/or merge files of any kind? I think `merging_check` should already work with non MC files, since there are already `if subarray_info0.name == "MonteCarloArray":` code checking this.

Reply:
Ha! My bad, you are absolutely right, we are using the merge to actually just create a new file.
In that case, I withdraw my comment, this is perfectly fine.
As for the refactoring, maybe that can go in another PR if you want to work on it, this one is perfectly fine as it is.